elliptic-curve/src/weierstrass/point.rs

```diff
@@ -171,6 +171,16 @@ where
     pub fn into_bytes(self) -> GenericArray<u8, UncompressedPointSize<C>> {
         self.bytes
     }
+
+    /// Get the x-coordinate of this curve point
+    pub(crate) fn x(&self) -> &ScalarBytes<C> {
```

Comment:
Something of a misuse of the `Scalar` part of `ScalarBytes` here... perhaps it should be renamed to `ElementBytes` to reflect the `Curve::ElementSize` rename (from `Curve::ScalarSize`)